src/BenchmarkDotNet/Disassemblers/IntelDisassembler.cs

```diff
@@ -53,27 +81,27 @@ protected override IEnumerable<Asm> Decode(byte[] code, ulong startAddress, Stat
 
                             FlushCachedDataIfNeeded(state.Runtime.DataTarget.DataReader, address, buffer);
 
-                            if (state.Runtime.DataTarget.DataReader.Read(address, buffer) == buffer.Length && buffer.SequenceEqual(callCountingStubTemplate))
+                            if (state.Runtime.DataTarget.DataReader.Read(address, buffer) == buffer.Length && buffer.SequenceEqual(data.callCountingStubTemplate))
                             {
                                 const ulong TargetMethodAddressSlotOffset = 8;
-                                address = state.Runtime.DataTarget.DataReader.ReadPointer(address + (ulong)Environment.SystemPageSize + TargetMethodAddressSlotOffset);
+                                address = state.Runtime.DataTarget.DataReader.ReadPointer(address + (ulong)data.stubPageSize + TargetMethodAddressSlotOffset);
                             }
                             else
                             {
                                 buffer = new byte[13];
-                                if (state.Runtime.DataTarget.DataReader.Read(address, buffer) == buffer.Length && buffer.SequenceEqual(stubPrecodeTemplate))
+                                if (state.Runtime.DataTarget.DataReader.Read(address, buffer) == buffer.Length && buffer.SequenceEqual(data.stubPrecodeTemplate))
                                 {
                                     const ulong MethodDescSlotOffset = 0;
-                                    address = state.Runtime.DataTarget.DataReader.ReadPointer(address + (ulong)Environment.SystemPageSize + MethodDescSlotOffset);
+                                    address = state.Runtime.DataTarget.DataReader.ReadPointer(address + (ulong)data.stubPageSize + MethodDescSlotOffset);
                                     isPrestubMD = true;
                                 }
                                 else
                                 {
                                     buffer = new byte[19];
-                                    if (state.Runtime.DataTarget.DataReader.Read(address, buffer) == buffer.Length && buffer.SequenceEqual(fixupPrecodeTemplate))
+                                    if (state.Runtime.DataTarget.DataReader.Read(address, buffer) == buffer.Length && buffer.SequenceEqual(data.fixupPrecodeTemplate))
                                     {
                                         const ulong MethodDescSlotOffset = 8;
-                                        address = state.Runtime.DataTarget.DataReader.ReadPointer(address + (ulong)Environment.SystemPageSize + MethodDescSlotOffset);
+                                        address = state.Runtime.DataTarget.DataReader.ReadPointer(address + (ulong)data.stubPageSize + MethodDescSlotOffset);
```

Comment:
nit: it seems that `stubPageSize` is casted to `ulong` with each use. We could make this field an `ulong` and avoid the need of casts.